src/main/java/org/folio/services/email/impl/MailServiceImpl.java

```diff
@@ -79,21 +82,38 @@ public void sendEmail(JsonObject configJson, JsonObject emailEntityJson, Handler
 
       defineMailClient(mailConfig)
         .sendMail(mailMessage, mailHandler -> {
+          boolean shouldRetry = false;
+          int retryCount = 0;
           if (mailHandler.failed()) {
+            Pair<Boolean, Integer> values = getRetryValues(mailHandler.cause(), emailEntity);
+            shouldRetry = values.getLeft();
+            retryCount = values.getRight();
             String errorMsg = String.format(ERROR_SENDING_EMAIL, mailHandler.cause().getMessage());
-            resultHandler.handle(fillResultHandler(emailEntity, Status.FAILURE, errorMsg));
+            resultHandler.handle(fillResultHandler(emailEntity, Status.FAILURE, errorMsg,
+              shouldRetry, retryCount));
             return;
           }
           // the logic of sending the result of sending email to `mod-notify`
           String message = createResponseMessage(mailHandler);
-          resultHandler.handle(fillResultHandler(emailEntity, Status.DELIVERED, message));
-        });
+          resultHandler.handle(fillResultHandler(emailEntity, Status.DELIVERED, message,
+            shouldRetry, retryCount));
+      });
     } catch (Exception ex) {
       logger.error(String.format(ERROR_SENDING_EMAIL, ex.getMessage()));
       resultHandler.handle(Future.failedFuture(ex.getMessage()));
     }
   }
 
+  private Pair<Boolean, Integer> getRetryValues(Throwable error, EmailEntity emailEntity) {
```

Comment:
```suggestion
  private boolean shouldRetry(Throwable error) {
```
You can get and increment `retryCount` in the caller-method. Also, you should not increment it after first sending attempt, only when you retry sending. Otherwise, we should not call it `retryCount`.